src/fondant/core/manifest.py

```diff
@@ -220,41 +195,76 @@ def cache_key(self) -> str:
         return self.metadata["cache_key"]
 
     @property
-    def index(self) -> Index:
-        return Index(self._specification["index"], base_path=self.base_path)
-
-    @property
-    def subsets(self) -> t.Mapping[str, Subset]:
+    def fields(self) -> t.Mapping[str, Field]:
         """The subsets of the manifest as an immutable mapping."""
+        # e.g. ('images', {'location': '/component1', 'type': 'binary'})
         return types.MappingProxyType(
             {
-                name: Subset(subset, base_path=self.base_path)
-                for name, subset in self._specification["subsets"].items()
+                name: Field(
+                    name=name,
+                    type=Type(field["type"]),
+                    location=field["location"],
+                )
+                for name, field in self._specification["fields"].items()
             },
         )
 
-    def add_subset(
+    def add_fields(
```

Comment:
This method is only used in tests as far as I can see, and `add_subset` is no longer used. So I think we can remove it.